framework/include/actions/CreateExecutionerAction.h

```diff
@@ -30,6 +30,9 @@ class CreateExecutionerAction : public MooseObjectAction
 
   virtual void act();
 
+  static MultiMooseEnum getCommonPetscOptions();
+  static MultiMooseEnum getCommonPetscOptionsIname();
```

Comment:
I really like this but lets move these static methods to a more logical location. How about utils/PetscSupport?